lib/Doctrine/deploy/AddProjectRoles.php

```diff
@@ -91,9 +88,13 @@
             $doctrineUser = $doctrineUser;
         }
 
-        $doctrineRole = new Role($roleType, $doctrineUser, $egi, 'STATUS_GRANTED');
-        $entityManager->persist($doctrineRole);
+        if ((string) $role->ON_ENTITY == "EGI") {
+            $doctrineRole = new Role($roleType, $doctrineUser, $egi, 'STATUS_GRANTED');
+            $entityManager->persist($doctrineRole);
+        } else {
+            continue;
+        }
```

Comment:
Can this be a look up on the underlying project entity and, regardless of whether it's EGI or another project, add the appropriate role?

Similar to https://github.com/rowan04/gocdb/blob/384-refresh_sample_data/lib/Doctrine/deploy/AddNGIRoles.php#L65-L78

Reply:
and ammend the docstring at the top of the file accordingly.